source/extensions/quic_listeners/quiche/platform/flags_impl.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <string>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/synchronization/mutex.h"
+
+namespace quiche {
+
+class Flag;
+
+// Registry of QUICHE flags. Can be used to reset all flags to default values,
```

Comment:
Hmmm.  So either now or later we're going to want to have this implemented based on Envoy's runtime, which is the Envoy canonical way of saying "I want to change this feature on a live binary".

Now that runtime is a singleton, the getter and setter code is absolutely trivial to implement for booleans/ints/strings for cross-compatibility.  The annoying bit would be getting a runtime set up for all QUIC unit tests.   I put some thought into how to do this for the "feature flag" work I did over in https://github.com/envoyproxy/envoy/pull/6134/files and while my final version didn't have a command line flag for "create a test registry per test case" I was playing around with it for an earlier iteration and it wasn't too terrible.

That said, I'm 100% fine with landing this with a TODO to do that work eventually.  Take a look and feel free to ping me for additional context, and you can sort out if it's worth the trouble now.